src/native/libs/System.Globalization.Native/pal_collation.m

```diff
@@ -0,0 +1,77 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <stdlib.h>
+#include "pal_locale_internal.h"
+#include "pal_collation.h"
+
+#import <Foundation/Foundation.h>
+
+#if defined(TARGET_OSX) || defined(TARGET_MACCATALYST) || defined(TARGET_IOS) || defined(TARGET_TVOS)
+
+// Enum that corresponds to C# CompareOptions
+typedef enum
+{
+    None = 0,
+    IgnoreCase = 1,
+    IgnoreNonSpace = 2,
+    IgnoreWidth = 16,
+} CompareOptions;
+
+#define CompareOptionsMask 0x1f
+
+static NSStringCompareOptions ConvertFromCompareOptionsToNSStringCompareOptions(int32_t comparisonOptions)
+{
+    int32_t supportedOptions = IgnoreCase | IgnoreNonSpace | IgnoreWidth;
+    NSStringCompareOptions options = NSLiteralSearch;
+    comparisonOptions &= CompareOptionsMask;
```

Comment:
ComparisonOptions.Ordinal should not reach here https://github.com/dotnet/runtime/blob/40438e009786fc10885f7ea4dbed185d5700a6aa/src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.cs#L444 
But instead having StringSort and mapping it will be more readable. Will update.